proposed/http-factory/http-factory.md

```diff
@@ -105,48 +116,44 @@ Has the ability to create streams for requests and responses.
 ```php
 namespace Psr\Http\Message;
 
+use Psr\Http\Message\StreamInterface;
+
 interface StreamFactoryInterface
 {
     /**
-     * Create a new stream with no content.
-     *
-     * The stream will be writable and seekable.
-     *
-     * @return StreamInterface
-     */
-    public function createStream();
-
-    /**
-     * Create a new stream from a callback.
+     * Create a new stream from a string.
      *
-     * The stream will be read-only and not seekable.
+     * The stream SHOULD be created with a temporary resource.
      *
-     * @param callable $callback
+     * @param string $content
      *
      * @return StreamInterface
      */
-    public function createStreamFromCallback(callable $callback);
+    public function createStream($content = '');
 
     /**
-     * Create a new stream from a resource.
+     * Create a stream from an existing file.
      *
-     * @param resource $body
+     * The file MUST be opened using the given mode, which may be any mode
+     * supported by the `fopen` function.
+     *
+     * @param string $file
+     * @param string $mode
      *
      * @return StreamInterface
      */
-    public function createStreamFromResource($body);
+    public function createStreamFromFile($file, $mode = 'r');
 
     /**
-     * Create a new stream from a string.
+     * Create a new stream from an existing resource.
      *
-     * A temporary resource will be created with the content of the string.
-     * The resource will be writable and seekable.
+     * The stream MUST be readable and may be writable.
```

Comment:
> Read-only streams are valid.

Yes, they are valid. But as an `interface` consumer, this information does not provide me any(?) useful information. E.g. the stream might be seekable as well, but as a consumer I cannot rely on that either.

But thinking about it, `may be writeable` explicitly allows writable streams, thus it might be useful for `interface` implementers. :+1: